src/coreclr/vm/syncblk.inl

```diff
@@ -723,7 +731,12 @@ FORCEINLINE AwareLock::LeaveHelperAction ObjHeader::LeaveObjMonitorHelper(Thread
         {
             // We are leaving the lock
             DWORD newValue = (syncBlockValue & (~SBLK_MASK_LOCK_THREADID));
+
+#if defined(TARGET_WINDOWS) && defined(TARGET_ARM64)
```

Comment:
I think moving the static/dynamic switching logic into `InterlockedCompareExchangeRelease` and similar would be nicer.

It should result in simpler looking code and if switching needs to change due to new compiler or platform, no need to revisit all the callsites.